integration/tests/access/cohort3/grpc_state_stream_test.go

```diff
@@ -330,44 +330,78 @@ func (s *GrpcStateStreamSuite) getRPCs() []RPCTest {
 	}
 }
 
-type ResponseTracker struct {
-	r  map[uint64]map[string]SubscribeEventsResponse
-	mu sync.RWMutex
+// ResponseTracker is a generic tracker for responses.
+type ResponseTracker[T any] struct {
+	r       map[uint64]map[string]T
+	mu      sync.RWMutex
+	compare func(t *testing.T, responses map[uint64]map[string]T, blockHeight uint64) error
 }
 
-func newResponseTracker() *ResponseTracker {
-	return &ResponseTracker{
-		r: make(map[uint64]map[string]SubscribeEventsResponse),
+// NewResponseTracker creates a new ResponseTracker.
+func NewResponseTracker[T any](
+	compare func(t *testing.T, responses map[uint64]map[string]T, blockHeight uint64) error,
+) *ResponseTracker[T] {
+	return &ResponseTracker[T]{
+		r:       make(map[uint64]map[string]T),
+		compare: compare,
 	}
 }
 
-func (r *ResponseTracker) Add(t *testing.T, blockHeight uint64, name string, events *SubscribeEventsResponse) {
+func (r *ResponseTracker[T]) Add(t *testing.T, blockHeight uint64, name string, response T) {
```

Comment:
I think we need a way to assert the tracker is done, which would make sure `r` is empty, and that some number of responses were checked.  something like 

```golang
func (r *ResponseTracker[T]) AssertAllResponsesHandled(t *testing.T, expectedCount int) {
	assert.Empty(t, r.r, "expected response cache to be empty")
	assert.Equal(t, expectedCount, r.checkCount)
}
```

Reply:
[cd6a7a3](https://github.com/onflow/flow-go/pull/5624/commits/cd6a7a3bab4e19f8b3c1e8b6591fc0672bc69edc)